submissions/tests/test_models.py

```diff
@@ -164,3 +163,39 @@ def test_highest_score_hidden(self):
         highest = ScoreSummary.objects.get(student_item=item).highest
         self.assertEqual(highest.points_earned, 1)
         self.assertEqual(highest.points_possible, 2)
+
+    def test_override_highest_no_score(self):
+        item = StudentItem.objects.create(
+            student_id="score_test_student",
+            course_id="score_test_course",
+            item_id="i4x://mycourse/special_presentation"
+        )
+
+        # Override score with no score
+        Score.create_override_score(item, 10, 9)
+        highest = ScoreSummary.objects.get(student_item=item).highest
+        self.assertEqual(highest.points_earned, 9)
+        self.assertEqual(highest.points_possible, 10)
+
+    def test_override_highest_w_score(self):
+        item = StudentItem.objects.create(
+            student_id="score_test_student",
+            course_id="score_test_course",
+            item_id="i4x://mycourse/special_presentation"
+        )
+        # Override score after a non-reset score
+        submission = Submission.objects.create(
+            student_item=item,
+            attempt_number=1,
+        )
+        Score.objects.create(
+            student_item=item,
+            submission=submission,
+            points_earned=1,
+            points_possible=15,
+        )
+
+        Score.create_override_score(item, 15, 5)
+        highest = ScoreSummary.objects.get(student_item=item).highest
```

Comment:
Scoring always takes the last score; whether or not the old implementation (coursewarestudentmodule) or the new implementation (Submissions repo). However, submissions has a built in submissions_scoresummary table that has foreign keys to highest and latest ids. Note in submissions, scores are immutable whereas csm only keeps the latest and previous are rolled off into coursestudentmodulehistory. That said, the summary table is being populated but not used anywhere just yet.